app/src/main/java/com/droidhats/campuscompass/viewmodels/MapViewModel.kt

```diff
@@ -1,24 +1,92 @@
 package com.droidhats.campuscompass.viewmodels
 
 import android.app.Application
+import androidx.fragment.app.FragmentActivity
 import androidx.lifecycle.AndroidViewModel
+import com.droidhats.campuscompass.models.Building
 import com.droidhats.campuscompass.models.Campus
+import com.droidhats.campuscompass.models.Map
 import com.droidhats.campuscompass.repositories.MapRepository
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.model.Polygon
 import java.io.InputStream
 
+
+/**
+ * A ViewModel for the map.
+ * Receives data from the MapRepository and Sends initialized map and other information to the MapFragment .
+ *
+ * @constructor Reads data from the map repository.
+ *
+ * @param application: The android view model interface requires that the (not null) main application be passed.
+ */
 class  MapViewModel(application: Application) : AndroidViewModel(application) {
 
     private val context = getApplication<Application>().applicationContext
     private var campuses: List<Campus>? = null
 
-    fun getCampuses(): List<Campus> = campuses!!
-
     // The activity is required to access the assets to open our json file where the info
     // is stored
     init {
         val inputStream: InputStream = context.assets.open("buildings.json")
         val json: String = inputStream.bufferedReader().use { it.readText() }
-        campuses = MapRepository.getInstance(json).getCampuses()
+        campuses = MapRepository.getInstance(context).getCampuses()
+    }
+
+
+    /**
+     * Returns a list of campus objects (SJW and Loyola).
+     */
+    fun getCampuses(): List<Campus> = campuses!!
+
```

Comment:
why double space here?

Reply:
Removed the double space in the new commit.